### DISCUSSION TITLE
Suggestions for UX in the type editor menu in Inspector

### DISCUSSION THREAD

**Discussion by CodeLazier** [godotengine/godot-proposals#6562] [GUI]
I define a "@export var data:Dictionary={}"  (key=String, value=MyRes) in a resource script, some inconvenient experience happens when I edit data in the editor, first the type menu is very long, I need to click "object", I think object becomes a submenu better, no need for secondary clicks, when I select object, I need to continue to select my custom class This is a longer list while user-defined global classes are in the last few, especially if their menus cannot yet drag scroll bars. I think the first entry in that menu should have a User Global-like submenu, or at least a secondary submenu like recent->  ,If only a small amount of data is OK, I have more than 1000 such keys to define:(

https://user-images.githubusercontent.com/14953625/227525121-3b210462-f1c7-49ad-adff-4e75e2ee1490.mp4

I also found a minor bug that deleting dictionary data would not restore the default edited entries

**Comment by Calinou** [comment]
See https://github.com/don-tnowe/godot-dictionary-inspector.

**Reply by CodeLazier** [reply]
I think the functionality provided by this proposal should be out-of-the-box, rather than through plugins or custom code. If it were just my case, I could quickly complete the task by writing my own tool to write all the data into Tres at once. But that's only for my specific situation. I believe one of the goals of Godot is to be out-of-the-box, and Godot provides too many useful features. I think there is still room for improvement in some user experience details. Of course, if this does not fit into Godot's current plans for changes or is not a high priority, I understand. And if there is anything inappropriate about my suggestion, I apologize.
